database/migrations/default/1742417174000_crash_ems_link/up.sql

```diff
@@ -0,0 +1,409 @@
+add column person_id integer unique deferrable initially deferred,
+add constraint ems__incidents_person_id foreign key (
+    person_id
+) references people (id);
+
+comment on column ems__incidents.person_id is 'Person ID matched to this record';
+
+--
+-- add column to track match metadata and multiple crash_pks
+--
+alter table ems__incidents
+add column crash_match_status text default 'unmatched'
+check (
+    crash_match_status in (
+        'unmatched',
+        'matched_by_automation',
+        'multiple_matches_by_automation',
+        'matched_by_manual_qa'
+    )
+),
+add column matched_crash_pks integer [],
+add constraint matched_crash_pks_non_null check (
+    (crash_match_status = 'multiple_matches_by_automation')
+    or (matched_crash_pks is NULL)
+);
+comment on column ems__incidents.crash_match_status is 'The status of the crash record match';
+comment on column ems__incidents.matched_crash_pks is 'The IDs of multiple crashes that were found to match this record. Can only be populated when crash_match_status is ''multiple_matches_by_automation''';
+
+--
+-- add column to assign injury severity
+--
+alter table ems__incidents
+add column patient_injry_sev_id integer generated always as (
+    case
+        when
+            lower(pcr_provider_impression_primary) = 'death on scene'
+            or lower(pcr_outcome) = 'deceased on scene'
+            or lower(pcr_patient_acuity_final)
+            = 'dead without resuscitation efforts (black)'
+            then 4
+        when
+            right(pcr_transport_priority, 1) = '3'
+            and lower(pcr_patient_acuity_final) = 'lower acuity (green)'
+            then 2
+        when
+            right(pcr_transport_priority, 1) = '3'
+            and lower(pcr_patient_acuity_final) = 'critical (red)'
+            then 1
+        when
+            right(pcr_transport_priority, 1) = '3'
+            and lower(pcr_patient_acuity_final) = 'emergent (yellow)'
+            then 1
+        when
+            right(pcr_transport_priority, 1) = '3'
+            and lower(pcr_patient_acuity_initial) = 'critical (red)'
+            then 1
+        when
+            right(pcr_transport_priority, 1) = '3'
+            and lower(pcr_patient_acuity_initial) = 'emergent (yellow)'
+            then 1
+        when
+            right(pcr_transport_priority, 1) = '3'
+            and lower(pcr_patient_acuity_level) = 'high acuity'
+            then 1
+        -- what is this pcr_transport_priority?
+        when
+            upper(left(pcr_transport_priority, 1)) = 'CHARLIE'
+            or upper(left(pcr_transport_priority, 1)) = 'D'
+            then 1
+        when
+            right(pcr_transport_priority, 1) = '3'
+            and lower(pcr_patient_acuity_initial) = 'lower acuity (green)'
+            then 2
+        when
+            upper(left(pcr_transport_priority, 1)) = 'minor injury'
+            or upper(left(pcr_transport_priority, 1)) = 'possible injury'
+            then 2
```

Comment:
i wanted to check in on my understand, but will this ever result in TRUE? since it is comparing a lowercase string of words to an uppercase single letter.

Reply:
you're totally right, thanks. i'm giving all of these cases a finally copy edit now 😵‍💫